components/shared/layout/lib.rs

```diff
@@ -59,6 +60,10 @@ use style_traits::CSSPixel;
 use webrender_api::units::{DeviceIntSize, LayoutPoint, LayoutVector2D};
 use webrender_api::{ExternalScrollId, ImageKey};
 
+pub fn shutdown_style_threads() {
+    StyleThreadPool::shutdown();
+}
+
```

Comment:
Instead of exposing a free function from layout that calls into the style API, please just call `StyleThreadPool::shutdown()`  from `libservo` and the Constellation.